presto-main/src/main/java/com/facebook/presto/sql/analyzer/Scope.java

```diff
@@ -185,13 +194,28 @@ private static boolean isColumnReference(QualifiedName name, RelationType relati
         return Optional.empty();
     }
 
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .addValue(relationId)
+                .toString();
+    }
+
     public static final class Builder
     {
+        private RelationId relationId = RelationId.anonymous();
```

Comment:
`Scope.create()` would fail then. Even if the check applied only when there's non-empty `RelationType` there would be places (e.g. in planner), where providing `relationId` is not possible.